src/System.Collections.Specialized/src/System/Collections/Specialized/OrderedDictionary.cs

```diff
@@ -174,18 +162,18 @@ private void EnsureObjectsTable()
                 {
                     throw new ArgumentOutOfRangeException(nameof(index));
                 }
-                EnsureObjectsArray();
-                EnsureObjectsTable();
-                object key = ((DictionaryEntry)_objectsArray[index]).Key;
-                _objectsArray[index] = new DictionaryEntry(key, value);
-                _objectsTable[key] = value;
+                ArrayList objectsArray = EnsureObjectsArray();
+                Hashtable objectsTable = EnsureObjectsTable();
+                object key = ((DictionaryEntry)objectsArray[index]!).Key;
```

Comment:
DictionaryEntry.Key is not nullable and somehow Debug.Assert(objectsArray[index] != null) is not working in this case, and objectsArray[index] would never be null so it is safe to bang here

Reply:
> somehow Debug.Assert(objectsArray[index] != null) is not working in this case

https://github.com/dotnet/roslyn/issues/34644